add generate-playbook utility

ktdreyer opened the pull request:
This can query a live Koji instance and generate the Ansible playbook YAML to recreate some or all of the tags.

mizdebsk commented on utils/generate-playbook:
```diff
@@ -0,0 +1,171 @@
+        session.listPackages(tag_name)
+        session.getTagExternalRepos(tag_name)
+    multicall_results = session.multiCall(strict=True)
+    session.multicall = False
```
You don't need to explicitly set session.multicall to False - multiCall() will do it by itself.

ktdreyer replied:
Thank you. I switched to the new multi-call interface in Koji 1.18 and this code is better now.

mizdebsk commented on utils/generate-playbook:
```diff
@@ -0,0 +1,171 @@
+    # Optionally filter our tag list by a regular expression.
+    if args.tag_regex:
+        regex = re.compile(args.tag_regex)
+        tags = [tag for tag in tags if regex.search(tag['name'])]
```
If regex is "foo" then it also matches tag called "foo-bar". Not sure if it was intended, but I wouldn't expect it myself.

ktdreyer replied:
Yeah, that is what I expected to happen. Thanks for calling this out, though. I've added this behavior to the `--help` text documentation.

mizdebsk commented on utils/generate-playbook:
```diff
@@ -0,0 +1,171 @@
+    args = parse_args()
+    session = common_koji.get_session(args.profile)
+
+    tags = session.listTags()
```
It would be nice to sort tags by name (by passing queryOpts, eg `listTags(queryOpts={'order': 'name'})`).

ktdreyer replied:
Totally agreed. I've added this suggestion in my branch.

mizdebsk commented on utils/generate-playbook:
```diff
@@ -0,0 +1,171 @@
+        regex = re.compile(args.tag_regex)
+        tags = [tag for tag in tags if regex.search(tag['name'])]
+
+    print('querying %i tags' % len(tags))
```
This message should go to stderr, otherwise it is mixed with the YAML output.

ktdreyer replied:
Great catch. Thank you. I've made this message go to stderr in my branch.

mizdebsk commented on utils/generate-playbook:
```diff
@@ -0,0 +1,171 @@
+    Quickly get all the information we need from Koji with as few XML-RPC
+    calls as possible.
+    """
+    # Note: This client multiCall method bombs out after about 35 minutes for
```
You could use "itercall", which is a wrapper around multiCall that makes it easier to use, and also splits huge calls into smaller chunks of configurable size. See Koji pull request: https://pagure.io/koji/pull-request/116

ktdreyer replied:
Thank you for pointing this out. I switched to the new multi-call interface in Koji 1.18 and I set `batch=100`.

ktdreyer commented:
Thank you for your review @mizdebsk ! I will make time to address these comments next week.

ktdreyer commented:
It took 12 months, but I have finally picked this utility up again. I rebased on the latest master and addressed your comments @mizdebsk . I've also added support for `koji_target` and I updated the regex feature so that it runs on the server-side now with Koji's `search` RPC.

This does require PyYAML 5.1 in order to pretty-print the YAML keys in a friendly order. That version of PyYAML is available in Fedora, but CentOS 8 only has 3.12-12.el8 in BaseOS.
